### PR TITLE
Adding reshape

### PR DESCRIPTION
**Pull request by drubinstein**

### CODE REVIEW

File: include/cotila/matrix/utility.h [hunk 1]
```diff
@@ -108,6 +108,15 @@ constexpr matrix<T, P, Q> submat(const matrix<T, M, N> &m, std::size_t i, std::s
     });
 }
 
+template <std::size_t P, std::size_t Q, std::size_t M, std::size_t N,
+          typename T, typename = std::enable_if<P * Q == M * N>>
+constexpr matrix<T, P, Q> reshape(const matrix<T, M, N> &m) {
+  return generate<P, Q>([&m]<std::size_t Q1 = Q, std::size_t N1 = N>(
+      std::size_t i, std::size_t j) {
+    return m[(i * Q1 + j) / N][(i * Q1 + j) % N];
```
Comment (calebzulawski) [inline include/cotila/matrix/utility.h]: Shouldn't reshape go column-wise, not row-wise?  Also, something isn't building on clang, if you look at the Travis logs.

Comment (drubinstein) [reply include/cotila/matrix/utility.h]: What do you mean by column-wise and not row-wise? I wrote it such that 
[[1,2,3,4]
 [5,6,7,8]]

would become
[[1,2],
 [3,4],
 [5,6],
 [7,8]]

Is that not what you expect?

Fixed the build btw.

Comment (drubinstein) [reply include/cotila/matrix/utility.h]: IPP MKL is also row major. I dont think they have anything like reshape though.

Comment (calebzulawski) [reply include/cotila/matrix/utility.h]: Fair enough.  I took a look at numpy as a reference, and they have an argument for choosing whether to use C- or Fortran-style indexing.  They chose C as the default, so this seems to be consistent with other libraries.

**Pull request merged**